### ISSUE TITLE
Feature Request: Update default prompt to use OSC 9;9

### ISSUE THREAD

**Issue by ghost** [PowerShell/PowerShell#15469]
tldr; change default pwsh prompt to use OSC 9;9, so that people can create a new tab in Windows Terminal with same working directory as current tab.

## Summary of the new feature/enhancement

wrt https://github.com/microsoft/terminal/issues/8166 

Windows Terminal recently added support for OSC 9;9 which is required for creating a new tab with the same working directory. How this works is that the prompt contains the path to the working directory wrapped in osc 9;9 which is used by the Windows Terminal to read the path and create new tab in same path.

Currently, people have to manually modify the powershell prompt to include OSC 9;9 like this https://gist.github.com/skyline75489/480d036db8ae9069b7009377e6eebb79

It would make it easier for people to make use of this feature if the default prompt for pwsh was updated to use OSC 9;9, then people won't require any manual changes, and things would "just work"

How to test the feature -
=========================
Open Windows Terminal with pwsh profile in a directory and use the above linked prompt
![image](https://user-images.githubusercontent.com/83939604/119833916-21fb3480-bf1d-11eb-993a-05f4801921ab.png)
Press Ctrl-Shift-D
The new tab that opens will be in the same working directory.
(If the prompt is not updated to use OSC 9;9, then it opens in the default directory of C:\Users\<user>

**Comment by iSazonov** [comment]
@axcadsd  Please add more info, repro steps and screenshots.

**Comment by 237dmitry** [comment]
What this sequence has to do?
```
$ "`e]9;9;$pwd`e\"
<Empty line>
$
```

**Comment by axcadsd** [comment]
@iSazonov i updated the issue, please check and let me know if any more information is required.

**Comment by 237dmitry** [comment]
This works without `Convert-Path`

**Comment by iSazonov** [comment]
@axcadsd Thanks for updates! I can confirm this feature works great!

**Comment by 237dmitry** [comment]
Not working if open new tab in Windows Terminal with command line.
```
wt -w 0 nt -p "PowerShell"  # profile name or
wt -w 0 nt pwsh -Nologo     # command
```

**Comment by rkeithhill** [comment]
Neat feature but keep in mind that such a change in PowerShell would not be seen by folks who use a module like `oh-my-posh` or `posh-git` that provide their own custom prompt functions.

**Comment by darkvertex** [comment]
> Neat feature but keep in mind that such a change in PowerShell would not be seen by folks who use a module like `oh-my-posh` or `posh-git` that provide their own custom prompt functions.

FYI _oh-my-posh_ **v3.89.1** released in February 2021 introduced OSC 9;9 support with the `"osc99": true` json config key (and works just fine in WSL, too!)
